packages/turbo-codemod/__tests__/clean-globs.test.ts

```diff
@@ -0,0 +1,133 @@
+      ["**/**/result.json", "**/result.json"],
+      ["**/**/**/**", "**"],
+      ["**/foo/**/**/bar/**", "**/foo/**/bar/**"],
+      ["**/foo/**/**/**/bar/**/**", "**/foo/**/bar/**"],
+      ["**/foo/**/**/**/**/bar/**/**/**", "**/foo/**/bar/**"],
+    ];
+
+    const { log } = getTransformerHelpers({
+      transformer: "test",
+      rootPath: ".",
+      options: { force: false, dry: false, print: false },
+    });
+
+    // Now let's test the function
+    badGlobPatterns.forEach(([input, output]) => {
+      expect(fixGlobPattern(input, log)).toBe(output);
+    });
+  });
+
+  test("doesn't update valid globs and prints a message", () => {
+    const { log } = getTransformerHelpers({
+      transformer: "test",
+      rootPath: ".",
+      options: { force: false, dry: false, print: false },
+    });
+
+    // Now let's test the function
+    expect(fixGlobPattern("a/b/c/*", log)).toBe("a/b/c/*");
+  });
+
+  test("transforms '!**/folder' to '**/[!folder]'", () => {
+    let badGlobPatterns = [
+      ["!**/dist", "**/[!dist]"],
+      ["!**/node_modules", "**/[!node_modules]"],
+      ["!**/foo/bar", "**/[!foo/bar]"],
+      ["!**/foo/bar/baz", "**/[!foo/bar/baz]"],
+      ["!**/foo/bar/baz/qux", "**/[!foo/bar/baz/qux]"],
+    ];
+
+    const { log } = getTransformerHelpers({
+      transformer: "test",
+      rootPath: ".",
+      options: { force: false, dry: false, print: false },
+    });
+
+    // Now let's test the function
+    badGlobPatterns.forEach(([input, output]) => {
+      expect(fixGlobPattern(input, log)).toBe(output);
+    });
+  });
+
+  test("transforms '**ext' to '**/*ext'", () => {
+    let badGlobPatterns = [
+      ["cypress/integration/**.test.ts", "cypress/integration/**/*.test.ts"],
+      ["scripts/**.mjs", "scripts/**/*.mjs"],
+      ["scripts/**.js", "scripts/**/*.js"],
+      ["src/types/generated/**.ts", "src/types/generated/**/*.ts"],
+      ["**md", "**/*md"],
+      ["**txt", "**/*txt"],
+      ["**html", "**/*html"],
+    ];
+
+    const { log } = getTransformerHelpers({
+      transformer: "test",
+      rootPath: ".",
+      options: { force: false, dry: false, print: false },
+    });
+
+    // Now let's test the function
+    badGlobPatterns.forEach(([input, output]) => {
+      expect(fixGlobPattern(input, log)).toBe(output);
+    });
+  });
+
+  test("transforms 'pre**' to pre*/**", () => {
+    let badGlobPatterns = [
+      ["pre**", "pre*/**"],
+      ["pre**/foo", "pre*/**/foo"],
+      ["pre**/foo/bar", "pre*/**/foo/bar"],
+      ["pre**/foo/bar/baz", "pre*/**/foo/bar/baz"],
+      ["pre**/foo/bar/baz/qux", "pre*/**/foo/bar/baz/qux"],
+    ];
+
+    const { log } = getTransformerHelpers({
+      transformer: "test",
+      rootPath: ".",
+      options: { force: false, dry: false, print: false },
+    });
+
+    // Now let's test the function
+    badGlobPatterns.forEach(([input, output]) => {
+      expect(fixGlobPattern(input, log)).toBe(output);
+    });
+  });
```

Comment:
Just because all of these transforms could happen in the same glob, I'd like to see some tests for truly horrible globs that hit all of the conditions e.g. `!foo**/**/**bar`